src/pages/Activity/index.js

```diff
@@ -9,120 +10,107 @@ import StyledActivity from './styles';
 
 export default function Activity(){
   const { courseId, chapterId, topicId, activityType, activityId } = useParams();
-  const { courseContext, setCourseContext } = useContext(CourseContext);
-  const [chapter, setChapter] = useState(null);
-  const [topic, setTopic] = useState(null);
-  const [theory, setTheory] = useState(null);
-  const [exercise, setExercise] = useState(null);
+  const { courseContext, setCourseContext, activities, setActivities } = useContext(CourseContext);
+  const [activity, setActivity] = useState(null);
+  const [progress, setProgress] = useState(null);
+  const [end, setEnd] = useState(false);
+  const [done, setDone] = useState(false);
   const history = useHistory();
   const token = localStorage.getItem('token');
 
   useEffect(() => {
     if (!courseContext || courseContext.id !== parseInt(courseId)) {
-      axios.get(`${process.env.API_BASE_URL}/clients/courses/${courseId}`, { headers: { 'X-Access-Token': token } })
+      axios.get(`${process.env.API_BASE_URL}/clients/courses/${courseId}/activities`, { headers: { 'X-Access-Token': token } })
       .then(resp => {
         setCourseContext(resp.data);
-        const chapterNeeded = resp.data.chapters.find(c => c.id === parseInt(chapterId));
-        setChapter(chapterNeeded);
-        const topicNeeded = chapterNeeded.topics.find(t => t.id === parseInt(topicId));
-        setTopic(topicNeeded);
-        if (activityType === 'theory'){
-          const activityNeeded = topicNeeded.theories.find(a => a.id === parseInt(activityId));
-          setTheory(activityNeeded);
-        } else {
-          const activityNeeded = topicNeeded.exercises.find(a => a.id === parseInt(activityId));
-          setExercise(activityNeeded);
-        }
       });
     } else {
-      const chapterNeeded = courseContext.chapters.find(c => c.id === parseInt(chapterId));
-      setChapter(chapterNeeded);
-      const topicNeeded = chapterNeeded.topics.find(t => t.id === parseInt(topicId));
-      setTopic(topicNeeded);
-      if (activityType === 'theory'){
-        const activityNeeded = topicNeeded.theories.find(a => a.id === parseInt(activityId));
-        setTheory(activityNeeded);
-      } else {
-        const activityNeeded = topicNeeded.exercises.find(a => a.id === parseInt(activityId));
-        setExercise(activityNeeded);
-      }
+      getActivity();
+      getProgress();
     }
   }, [courseId, chapterId, topicId, activityType, activityId]);
 
-  function nextUrl(course, chapter, topic, type, activity){
-    const url = `/courses/${course}/chapter/${chapter}/topic/${topic}/${type}/${activity}`;
-    return url;
+  useEffect(() => {
+    if (activities) getActivity();
+    if (activities) getProgress();
```

Comment:
Aqui é a essa msm condição mesmo? Se for será que não poderia colocar dentro do mesmo escopo? Tipo: 
`if (activities) {
  getActivity();
  getProgress();
}`